libs/media-player/src/audio-annotate.cy.tsx

```diff
@@ -0,0 +1,234 @@
+            });
+        });
+
+        describe(`when the media has been played`, () => {
+            beforeEach(() => {
+                cy.get('audio').then(([audioElement]) => {
+                    audioElement.play();
+                });
+            });
+
+            describe(`the mark in-point button`, () => {
+                it(`should be enabled`, () => {
+                    cy.getByDataAttribute('in-point-marker-button').should('be.enabled');
+                });
+            });
+
+            describe(`the mark out-point button`, () => {
+                it(`should be disabled`, () => {
+                    cy.getByDataAttribute('in-point-marker-button').should('be.disabled');
+                });
+            });
+
+            describe(`when the mark in-point button is clicked`, () => {
+                it(`should mark an in-point`, () => {
+                    cy.wait(2000);
+
+                    cy.getByDataAttribute('in-point-marker-button').click();
+                });
+            });
+
+            describe(`when the mark in-point button is clicked, the mark out-point button`, () => {
+                it('should be enabled', () => {
+                    cy.wait(2000);
+
+                    cy.getByDataAttribute('in-point-marker-button').click();
+
+                    cy.wait(2300);
+
+                    cy.getByDataAttribute('out-point-marker-button').should('be.enabled');
+                });
+            });
+
+            describe(`when the mark in-point button is clicked, clicking the mark out-point button`, () => {
+                it('should mark an out-point', () => {
+                    cy.wait(2000);
+
+                    cy.getByDataAttribute('in-point-marker-button').click();
+
+                    cy.wait(2500);
+
+                    cy.getByDataAttribute('out-point-marker-button').click();
+
+                    cy.getByDataAttribute('time-range-selected').should('exist');
+
+                    // If I remove this cy.wait() the test suite fails at the first click
+                    cy.wait(2000);
+                });
+            });
+
+            describe(`when the mark in-point button is clicked and the user scrubs the playhead back before the in-point, clicking the mark out-point button`, () => {
+                it('should fail with an error', () => {
+                    cy.wait(4000);
+
+                    cy.getByDataAttribute('in-point-marker-button').click();
+
+                    cy.get('audio').then(([audioElement]) => {
+                        audioElement.currentTime = 1.0;
+                    });
+
+                    cy.getByDataAttribute('out-point-marker-button').click();
+
+                    cy.getByDataAttribute('time-range-selected').should('not.exist');
+
+                    cy.getByDataAttribute('audio-error-message').should('exist');
+
+                    // If I remove this cy.wait() the test suite fails at the first click
+                    cy.wait(2000);
```

Comment:
Yea. I'm not a big fan of the hacky approach to async (race conditions are possible), but this is tricky business, so I'm ok with some pragmatism here.

Reply:
put one `wait()` call for 1500 ms in the `beforeEach()` to replace any inline calls that we're only for hacking.  the other calls in the tests and now only for the specific timing of breakpoints.